### ISSUE TITLE
Integrate laravel migrations in MODX

### ISSUE THREAD

**Issue by sdrenth** [modxcms/revolution#14343]
### Summary
I would like to know what people think of the idea of possibly integrating Laravel Migrations into the core of MODX. 

### Why is it needed?
Laravel migrations is a nice, simple and clean way for creating/altering the database and also makes it possible to revert database changes.

I think it would also come in handy for developers when working with MODX they can easily create a database migration, push to staging or production and run the migrations.

Also it keeps track of the migrations that already have been processed by using a custom migrations table.

If there is enough support for this idea, then maybe someone can pickup this idea later on to actually integrate this into the core. For now, this is just an idea and wanted to know what people think about this.

**Comment by Mark-H** [comment]
"Should we integrate {$framework}" is putting the cart before the horse. We have APIs (xPDOManager) for database migrations that are used by core and extras. Apparently, those are not sufficient for your needs. Let's talk about the problem before jumping to a specific implementation to fix them.

**Comment by sdrenth** [comment]
@Mark-H You're right about that, we might as well use the same approach as Laravel Migration but using MODX APIs for it.

I think the general approach of laravel migrations would be very useful when developing in multiple environments (local/staging/production) and could be used for installing MODX core and making it also useful for developers when developing with MODX.

The migrations table comes in very handy to have a sort of "log" of which migrations have been completed.

**Comment by JoshuaLuckers** [comment]
I prefer [Phinx](https://phinx.org) fro migrations. Maybe it's better to keep the freedom of choice and come up with ideas and solutions to make it possible to use whatever tool a developer prefers for migrations.
